plugins/arElasticSearchPlugin/lib/model/arElasticSearchAccession.class.php

```diff
@@ -51,38 +57,57 @@ public function populate()
     return $errors;
   }
 
-  public static function serialize($object)
+  private static function serialize($id)
   {
-    $serialized = array();
+    if (!isset(self::$conn))
+    {
+      self::$conn = Propel::getConnection();
+    }
+
+    if (!isset(self::$statements['accession']))
+    {
+      $sql = 'SELECT acc.*, slug.slug
+        FROM '.QubitAccession::TABLE_NAME.' acc
+        JOIN '.QubitSlug::TABLE_NAME.' slug ON acc.id = slug.object_id
+        WHERE acc.id = :id';
 
-    $serialized['id'] = $object->id;
-    $serialized['slug'] = $object->slug;
+      self::$statements['accession'] = self::$conn->prepare($sql);
+    }
 
-    $serialized['identifier'] = $object->identifier;
+    self::$statements['accession']->execute(array(':id' => $id));
+    $data = self::$statements['accession']->fetch(PDO::FETCH_ASSOC);
 
-    $serialized['date'] = arElasticSearchPluginUtil::convertDate($object->date);
-    $serialized['createdAt'] = arElasticSearchPluginUtil::convertDate($object->createdAt);
-    $serialized['updatedAt'] = arElasticSearchPluginUtil::convertDate($object->updatedAt);
+    if (false === $data)
+    {
+      throw new sfException("Couldn't find accession (id: $id)");
```

Comment:
This probably should never occur, but if it does, is it better to throw an exception and stop the index process, or log the error and keep going?

Reply:
Thanks @sbreker, I've implemented this like we do for the other resources. If the exception is thrown in the populate task it's caught (in the populate method above) and logged at the end. In a single accession indexing (update method below) the exception won't be caught and it will cause an error.

I'm not a big fan of the implementation either, specially the exception message in the ES IO PDO class, where it can be thrown for various reasons, but I think we should addressed it in a different ticket/PR for all resources if it's needed.

Reply:
Ok that's great - Thanks @jraddaoui!